### PR TITLE
Add/todolist3

### PR DESCRIPTION
**Pull request by manitheleo**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,75 +1,76 @@
...
+    )
+    :
+    (
+    <Form
+    handleFormSubmit={handleFormSubmit} 
+    todo={todo} 
+    handleInputChange={handleInputChange}
+    />
+    )
+    }
```
Comment (ArashKianiamphora) [inline src/App.js:68]: look for a way to turn this into a single component

File: src/App.js [hunk 2]
```diff
@@ -1,75 +1,76 @@
...
+    setTasks(updateItem);
+  }
+  const handleEditClick = (todo, index) => {
+    setEdit(true);
+    setCurrentTodo({ ...todo });
+  }
+  const deleter = (value) => {
+    const del = tasks.filter((fine, index) => index!==value);
+    setTasks(del);
+  };
```
Comment (ArashKianiamphora) [inline src/App.js:49]: handleEditClick is a nice name, deleter makes no sense please be consistent when you name things

File: src/Components/Buttons/UpdateButton.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Button } from "@mui/material"
+const UpdateButton = () => {
+    return (
+        <>
+         <Button type="submit">Update</Button>
+        </>
```
Comment (ArashKianiamphora) [inline src/Components/Buttons/UpdateButton.js:6]: why are you using fragment ? why are you adding type ?

Please stop copying from internet and try to understand why you do what you do

File: src/Components/Buttons/CancelButton.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Button } from "@mui/material"
+const CancelButton = (props) => {
+    return (
+        <>
+         <Button onClick={()=>props.setEdit(false)}>cancel</Button>
+        </>
+    )
+}
+export default CancelButton
```
Comment (ArashKianiamphora) [inline src/Components/Buttons/CancelButton.js:9]: these componants are not really dont anything please split your code in a different way

Comment (ArashKianiamphora) [reply src/Components/Buttons/CancelButton.js:9]: by components, i mean CancelButton, DeleteButton and EditButton

File: src/App.js (hunk 1 above)
Comment (ArashKianiamphora) [reply src/App.js:68]: try to fix the indentations brother... how are you even reading this?

File: src/App.js [hunk 5]
```diff
@@ -1,75 +1,76 @@
...
+  const handleEditFormSubmit = (e) => {
+    e.preventDefault();
+    handleUpdateTodo(currentTodo.id, { ...currentTodo })
+  }
+  const handleUpdateTodo = (id, updateTodo) => {
+    const updateItem=tasks.map((todo,index) => {
+      return todo.id===id ? updateTodo : todo
+    })
+    setEdit(false);
+    setTasks(updateItem);
```
Comment (ArashKianiamphora) [inline src/App.js:40]: when changing state the correct way to set values in useState here is by doing this

```javascript
setTasks((...prevTasks)=> {
   return prevTasks.map((todo)=> todo.id===id ? updateTodo : todo)
})
````

File: src/Components/Buttons/UpdateButton.js (hunk 3 above)
Comment (manitheleo) [reply src/Components/Buttons/UpdateButton.js:6]: this is onsubmit functionality for form

**Pull request merged**